folium/plugins/timedynamic_geo_json.py

```diff
@@ -0,0 +1,257 @@
+    --------
+    >>> # Providing file that shall be embedded.
+    >>> GeoJson(open('foo.json'))
+    >>> # Providing filename that shall not be embedded.
+    >>> GeoJson('foo.json')
+    >>> # Providing dict.
+    >>> GeoJson(json.load(open('foo.json')))
+    >>> # Providing string.
+    >>> GeoJson(open('foo.json').read())
+
+    >>> # Provide a style_function that color all states green but Alabama.
+    >>> style_function = lambda x: {'fillColor': '#0000ff' if
+    ...                             x['properties']['name']=='Alabama' else
+    ...                             '#00ff00'}
+    >>> GeoJson(geojson, style_function=style_function)
+
+    """
+    def __init__(self, data, styledict, style_function=None, name=None,
+                 overlay=True, control=True, smooth_factor=None,
+                 highlight_function=None):
+        super(TimeDynamicGeoJson, self).__init__(name=name, overlay=overlay,
+                                                control=control)
+        self._name = 'GeoJson'
+        if hasattr(data, 'read'):
+            self.embed = True
+            self.data = json.load(data)
+        elif isinstance(data, dict):
+            self.embed = True
+            self.data = data
+        elif isinstance(data, text_type) or isinstance(data, binary_type):
+            if data.lstrip()[0] in '[{':  # This is a GeoJSON inline string
+                self.embed = True
+                self.data = json.loads(data)
+            else:  # This is a filename
+                self.embed = False
+                self.data = data
+        elif data.__class__.__name__ in ['GeoDataFrame', 'GeoSeries']:
+            self.embed = True
+            if hasattr(data, '__geo_interface__'):
+                # We have a GeoPandas 0.2 object.
+                self.data = json.loads(json.dumps(data.to_crs(epsg='4326').__geo_interface__))  # noqa
+            elif hasattr(data, 'columns'):
+                # We have a GeoDataFrame 0.1
+                self.data = json.loads(data.to_crs(epsg='4326').to_json())
+            else:
+                msg = 'Unable to transform this object to a GeoJSON.'
+                raise ValueError(msg)
+        else:
+            raise ValueError('Unhandled object {!r}.'.format(data))
+
+        self.styledict = styledict
+
+        # make set of timestamps
+        self.timestamps = set()
+        for feature in self.styledict.values():
+            self.timestamps.update(set(feature.keys()))
+        self.timestamps = sorted(list(self.timestamps))
+
+        if style_function is None:
+            def style_function(x):
+                return {}
+
+        self.style_function = style_function
+
+        self.highlight = highlight_function is not None
+
+        if highlight_function is None:
+            def highlight_function(x):
+                return {}
+
+        self.highlight_function = highlight_function
+
+        self.smooth_factor = smooth_factor
+
+        self._template = Template(u"""
```

Comment:
You can probably inherit from the GeoJson class and save a lot of the duplication by only changing the template.

Reply:
@ocefpaf Good idea!